### DISCUSSION TITLE
What's the downside of using a long timeout on endpoint.read() (barcode scanner)

### DISCUSSION THREAD

**Discussion by samshipengs** [pyusb/pyusb#473] [Q&A] [answered]
I'm currently trying to get scanned values from a barcode scanner. The rough script I'm using is the following:

```python
while True:
    try:
        # Wait up to 0.5 seconds for data. 500 = 0.5 second timeout.
        data = ep.read(1000, 500)  
        
        Some Operation here that takes for example 2s ...
    except usb.core.USBError:
        # Timed out. End of the data stream. Print the scan line.
        if len(line) > 0:
            print line
            line = ''
```

With 500ms (0.5s) timeout, I often get incomplete scanned values from `ep.read`, for example, I'm scanning a 10 digits long code and with 0.5s, I often get less than 10 digits (e.g. 6 digits) in one `try` and then the rest digits (4 digits) would come in the next while iteration. I think the reason is that, when I scan the barcode, the python script could be in any stage, it might be in the half mark of the `0.5s` timeout, then there is not enough time left for the `ep.read` to pick up all the data packets, it could just pick up partial.

I'd like to capture all 10 digits in one `ep.read` as much as possible, what I can think of is that, I can increase the timeout to a large value e.g. 30s, this would almost ensure `ep.read` would get enough time to read the data packet. However, is there any downside if I increase the timeout to such value?

**Comment by mcuee** [comment, marked as answer]
Maybe you can try 5s first to see if that helps. If that does not work then increase the timeout valaue. It should be okay to increase the timeout value in your use case.
